src/server/server/frontendstructs.py

```diff
@@ -46,15 +46,17 @@ def __init__(self, document, grade):
 class User(object):
 
     @staticmethod
-    def make_from_db(email, user_doc_set_db_key, feature_vector_db_key):
-        return User(email, user_doc_set_db_key, feature_vector_db_key)
+    def make_from_db(email, password, interests, user_doc_set_db_key, feature_vector_db_key):
+        return User(email, password, interests, user_doc_set_db_key, feature_vector_db_key)
 
     @staticmethod
-    def make_from_scratch(email):
-        return User(email, None, None)
+    def make_from_scratch(email, password, interests):
+        return User(email, password, interests, None, None)
 
-    def __init__(self, email, user_doc_set_db_key, feature_vector_db_key):
+    def __init__(self, email, password, interests, user_doc_set_db_key, feature_vector_db_key):
         self.email = email
+        self.password = password
+        self.interests = interests
         self._user_doc_set_db_key = user_doc_set_db_key
```

Comment:
For now "interests" is one string, I think its more intuitive (a suggested by naming) if it's a list of strings. It's certainly convenient now because we guess we will just pass it as such to the topicmodeller, but it's a implementaiton detail of the module that would be "FeatureExtractorFromInterests" to merge them back. We don't want to assume in the database a "separator" token as ";" or " ". ie: "computer science" is ONE interest. Or the user can fill a different box for each different interest.

By splitting explicitly, we also keep the possibility to index in the future and to request database on one interest, which could ne usefull.